gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsException.java

```diff
@@ -31,16 +31,16 @@
 
   // see: https://cloud.google.com/dns/troubleshooting
   private static final Set<Error> RETRYABLE_ERRORS = ImmutableSet.of(
-      new Error(429, null),
-      new Error(500, null),
-      new Error(502, null),
-      new Error(503, null),
-      new Error(null, "userRateLimitExceeded"),
-      new Error(null, "rateLimitExceeded"));
+      new Error(429, null, true),
+      new Error(500, null, false),
+      new Error(502, null, false),
+      new Error(503, null, false),
+      new Error(null, "userRateLimitExceeded", true),
```

Comment:
can these reasons be part of the previous error codes? If so, this is why you made the loop in Error iterate over all values?